### DISCUSSION TITLE
AutoGen Extended

### DISCUSSION THREAD

**Discussion by jmsbooth** [microsoft/autogen#118] [Ideas]
Idea is to be able to generate individual **.py files as an individual team with multiple agents, including a manager. Then create multiple of these specialized teams and then a central management team that owns the strategic planning for all of the other teams.

Right now, I don't see a good way to pass these chats across from team to team that are running in separate processes. A message bus could work, using the output of the current chat to a file and then passing that file as an input to the other team. Then how to get the responses from the agents to WAIT till they receive a response from the other team?

Alternatively, simply wrapping the agents into a container with the ability to have a config, entry, and exit points so that each container can be spun up as a customized team  and then scale laterally as needed. 

Essentially creating a fully automated enterprise.... dev, marketing, sales, QA, logistics, business development, etc. etc.

**Comment by sonichi** [comment]
One way is to use function call. Example: https://github.com/microsoft/autogen/blob/main/notebook/agentchat_planning.ipynb
